plugins/de.fraunhofer.ipa.ros.xtext/src/de/fraunhofer/ipa/ros/Ros.xtext

```diff
@@ -6,221 +6,282 @@ import "http://www.ipa.fraunhofer.de/primitives" as primitives
-	'}';
+@Override 
+terminal SL_COMMENT: '#' !('\n'|'\r')*;
 
+///////////////////
+// CONVENTIONS AND NAMES
+///////////////////
 EString returns ecore::EString:
 	STRING | ID;
 	
 RosNames returns ecore::EString:
 	ROS_CONVENTION_A | ID | 'node'
 ;
 
-RosParamNames returns ecore::EString:
-	ROS_CONVENTION_PARAM | ID 
-;
-
 terminal ROS_CONVENTION_A:
 	( ('/' ID ) | ( ID '/' ) )* ;
 
 terminal ROS_CONVENTION_PARAM:
 	( ('/' STRING ) | ( STRING '/' ) | ('~' STRING ) )* ;
-	
-Artifact returns Artifact:
-	{Artifact}
-	'Artifact'
-	name=RosNames
-	'{'
-		(node=Node)?
-	'}';
+
+///////////////////
+//PACKAGES
+///////////////////
+Package returns Package:
+	Package_Impl //|
+	//CatkinPackage | 
+	//AmentPackage
+	;
+
+Package_Impl returns Package:
+	{Package}
+	name=RosNames':'
+	BEGIN
+		('fromGitRepo:' fromGitRepo=EString)?
+		('specs:' 
+			BEGIN
+			spec+=SpecBase*
+			END
+		)?
+		('dependencies:' '[' dependency+=Dependency (',' dependency+=Dependency)* ']' )?
+	END;
 
 CatkinPackage returns CatkinPackage:
 	{CatkinPackage}
-	'CatkinPackage'
-	name=RosNames
-	'{'
-		('FromGitRepo' fromGitRepo=EString)?
-		('Dependencies' '{' dependency+=Dependency ( "," dependency+=Dependency)* '}' )?		
-		('Specs'  '{' spec+=SpecBase ( "," spec+=SpecBase)* '}')?
-		(artifact+=Artifact ( "," artifact+=Artifact)*)?
-	'}';
-	
+	name=RosNames':'
+	BEGIN
+		('fromGitRepo:' fromGitRepo=EString)?
+		('artifacts:' 
+			BEGIN 
+			artifact+=Artifact*
+			END
+		)?
+		('dependencies:' '[' dependency+=Dependency (',' dependency+=Dependency)* ']' )?
+	END;
+
 AmentPackage returns AmentPackage:
 	{AmentPackage}
 	'AmentPackage'
-	name=RosNames
-	'{'
-		('FromGitRepo' fromGitRepo=EString)?
-		('Dependencies' '{' dependency+=Dependency ( "," dependency+=Dependency)* '}' )?		
-		('Specs'  '{' spec+=SpecBase ( "," spec+=SpecBase)* '}')?
-		(artifact+=Artifact ( "," artifact+=Artifact)*)?
-	'}';
+	name=RosNames':'
+	BEGIN
+		('fromGitRepo:' fromGitRepo=EString)?
+		('artifacts:' 
+			BEGIN 
+			artifact+=Artifact*
+			END
+		)?
+		('dependencies:' '[' dependency+=Dependency (',' dependency+=Dependency)* ']' )?
+	END;
+
+///////////////////
+//ARTIFACT AND NODE
+///////////////////
+Artifact returns Artifact:
+	{Artifact}
+		name=RosNames':'
+		BEGIN
+		(node=Node)?
+		END;
 
-ServiceSpec returns ServiceSpec:
-	{ServiceSpec}
-	'ServiceSpec'
-	name=EString
-	'{'
-		('request' request=MessageDefinition)?
-		('response' response=MessageDefinition)?
-	'}';
+Node returns Node:
+	'node:' name=RosNames
+	 BEGIN
+		(
+		('publishers:' 
+			BEGIN 
+			publisher+=Publisher*
+			END
+		)|
+		('subscribers:' 
+			BEGIN 
+			subscriber+=Subscriber*
+			END
+		)|
+		('serviceserver:' 
+			BEGIN 
+			serviceserver+=ServiceServer*
+			END
+		)|
+		('serviceclient:' 
+			BEGIN 
+			serviceclient+=ServiceClient*
+			END
+		)|
+		('actionserver:' 
+			BEGIN 
+			actionserver+=ActionServer*
+			END
+		)|
+		('actionclient:' 
+			BEGIN 
+			actionclient+=ActionClient*
+			END
+		)|
```

Comment:
Probably make this plural.
```suggestion
		('serviceservers:' 
			BEGIN 
			serviceserver+=ServiceServer*
			END
		)|
		('serviceclients:' 
			BEGIN 
			serviceclient+=ServiceClient*
			END
		)|
		('actionservers:' 
			BEGIN 
			actionserver+=ActionServer*
			END
		)|
		('actionclients:' 
			BEGIN 
			actionclient+=ActionClient*
			END
		)|
```

Reply:
Thanks a lot for the suggestion, I will add this change. It makes sense! :)